[🐛 Bug]: flaky elastic integration test

```
FAILED tests/test_enrichments.py::test_enrichment_with_elastic[NO_AUTH] - Exception: Failed to run query in Elastic: BadRequestError(400, 'verification_exception', 'Found 1 problem\nline 1:44: Unknown column [lastReceived]')
===== 1 failed, 88 passed, 736 skipped, 344 warnings in 167.77s (0:02:47) ======
```

```
=================================== FAILURES ===================================
____________________ test_enrichment_with_elastic[NO_AUTH] _____________________
  + Exception Group Traceback (most recent call last):
  |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/_utils.py", line 76, in collapse_excgroups
  |     yield
  |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 186, in __call__
  |     async with anyio.create_task_group() as task_group:
  |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 815, in __aexit__
  |     raise BaseExceptionGroup(
  | ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
  +-+---------------- 1 ----------------
    | Traceback (most recent call last):
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/_pytest/runner.py", line 341, in from_call
    |     result: TResult | None = func()
    |                              ^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/_pytest/runner.py", line 242, in <lambda>
    |     lambda: runtest_hook(item=item, **kwds), when=when, reraise=reraise
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/pluggy/_hooks.py", line 513, in __call__
    |     return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/pluggy/_manager.py", line 120, in _hookexec
    |     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/pluggy/_callers.py", line 182, in _multicall
    |     return outcome.get_result()
    |            ^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/pluggy/_result.py", line 100, in get_result
    |     raise exc.with_traceback(exc.__traceback__)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/pluggy/_callers.py", line 167, in _multicall
    |     teardown.throw(outcome._exception)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/_pytest/threadexception.py", line [92](https://github.com/keephq/keep/actions/runs/15237037962/job/42853209989?pr=4904#step:8:93), in pytest_runtest_call
    |     yield from thread_exception_runtest_hook()
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/_pytest/threadexception.py", line 68, in thread_exception_runtest_hook
    |     yield
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/pluggy/_callers.py", line 167, in _multicall
    |     teardown.throw(outcome._exception)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/_pytest/unraisableexception.py", line 95, in pytest_runtest_call
    |     yield from unraisable_exception_runtest_hook()
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/_pytest/unraisableexception.py", line 70, in unraisable_exception_runtest_hook
    |     yield
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/pluggy/_callers.py", line 167, in _multicall
    |     teardown.throw(outcome._exception)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/_pytest/logging.py", line 846, in pytest_runtest_call
    |     yield from self._runtest_for(item, "call")
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/_pytest/logging.py", line 829, in _runtest_for
    |     yield
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/pluggy/_callers.py", line 167, in _multicall
    |     teardown.throw(outcome._exception)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/_pytest/capture.py", line 880, in pytest_runtest_call
    |     return (yield)
    |             ^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/pluggy/_callers.py", line 167, in _multicall
    |     teardown.throw(outcome._exception)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/_pytest/skipping.py", line 257, in pytest_runtest_call
    |     return (yield)
    |             ^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/pluggy/_callers.py", line 103, in _multicall
    |     res = hook_impl.function(*args)
    |           ^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/_pytest/runner.py", line 174, in pytest_runtest_call
    |     item.runtest()
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/_pytest/python.py", line 1627, in runtest
    |     self.ihook.pytest_pyfunc_call(pyfuncitem=self)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/pluggy/_hooks.py", line 513, in __call__
    |     return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/pluggy/_manager.py", line 120, in _hookexec
    |     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/pluggy/_callers.py", line 182, in _multicall
    |     return outcome.get_result()
    |            ^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/pluggy/_result.py", line 100, in get_result
    |     raise exc.with_traceback(exc.__traceback__)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/pluggy/_callers.py", line 103, in _multicall
    |     res = hook_impl.function(*args)
    |           ^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/_pytest/python.py", line 159, in pytest_pyfunc_call
    |     result = testfunction(**testargs)
    |              ^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/tests/test_enrichments.py", line 462, in test_enrichment_with_elastic
    |     response = client.get(
    |                ^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/testclient.py", line 514, in get
    |     return super().get(
    |            ^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/httpx/_client.py", line 1055, in get
    |     return self.request(
    |            ^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/testclient.py", line 484, in request
    |     return super().request(
    |            ^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/httpx/_client.py", line 828, in request
    |     return self.send(request, auth=auth, follow_redirects=follow_redirects)
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/httpx/_client.py", line 915, in send
    |     response = self._send_handling_auth(
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/httpx/_client.py", line [94](https://github.com/keephq/keep/actions/runs/15237037962/job/42853209989?pr=4904#step:8:95)3, in _send_handling_auth
    |     response = self._send_handling_redirects(
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/httpx/_client.py", line 980, in _send_handling_redirects
    |     response = self._send_single_request(request)
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/httpx/_client.py", line 1016, in _send_single_request
    |     response = transport.handle_request(request)
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/testclient.py", line 377, in handle_request
    |     raise exc
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/testclient.py", line 374, in handle_request
    |     portal.call(self.app, scope, receive, send)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/anyio/from_thread.py", line 290, in call
    |     return cast(T_Retval, self.start_task_soon(func, *args).result())
    |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/concurrent/futures/_base.py", line 456, in result
    |     return self.__get_result()
    |            ^^^^^^^^^^^^^^^^^^^
    |   File "/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/concurrent/futures/_base.py", line 401, in __get_result
    |     raise self._exception
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/anyio/from_thread.py", line 221, in _call_func
    |     retval = await retval_or_awaitable
    |              ^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/fastapi/applications.py", line 1054, in __call__
    |     await super().__call__(scope, receive, send)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/applications.py", line 113, in __call__
    |     await self.middleware_stack(scope, receive, send)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/middleware/errors.py", line 187, in __call__
    |     raise exc
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/middleware/errors.py", line 165, in __call__
    |     await self.app(scope, receive, _send)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/opentelemetry/instrumentation/asgi/__init__.py", line 743, in __call__
    |     await self.app(scope, otel_receive, otel_send)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 185, in __call__
    |     with collapse_excgroups():
    |   File "/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/contextlib.py", line 158, in __exit__
    |     self.gen.throw(typ, value, traceback)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/_utils.py", line 82, in collapse_excgroups
    |     raise exc
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 187, in __call__
    |     response = await self.dispatch_func(request, call_next)
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/keep/api/observability.py", line 107, in __call__
    |     response = await call_next(request)
    |                ^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 163, in call_next
    |     raise app_exc
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 149, in coro
    |     await self.app(scope, receive_or_disconnect, send_no_error)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/prometheus_fastapi_instrumentator/middleware.py", line 174, in __call__
    |     raise exc
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/prometheus_fastapi_instrumentator/middleware.py", line 172, in __call__
    |     await self.app(scope, receive, send_wrapper)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 185, in __call__
    |     with collapse_excgroups():
    |   File "/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/contextlib.py", line 158, in __exit__
    |     self.gen.throw(typ, value, traceback)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/_utils.py", line 82, in collapse_excgroups
    |     raise exc
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 187, in __call__
    |     response = await self.dispatch_func(request, call_next)
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/keep/api/middlewares.py", line 60, in dispatch
    |     response = await call_next(request)
    |                ^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 163, in call_next
    |     raise app_exc
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 149, in coro
    |     await self.app(scope, receive_or_disconnect, send_no_error)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/middleware/cors.py", line 85, in __call__
    |     await self.app(scope, receive, send)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/middleware/gzip.py", line 20, in __call__
    |     await responder(scope, receive, send)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/middleware/gzip.py", line 39, in __call__
    |     await self.app(scope, receive, self.send_with_gzip)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette_context/middleware/raw_middleware.py", line 92, in __call__
    |     await self.app(scope, receive, send_wrapper)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
    |     await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    |     raise exc
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
    |     await app(scope, receive, sender)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/routing.py", line 715, in __call__
    |     await self.middleware_stack(scope, receive, send)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/routing.py", line 735, in app
    |     await route.handle(scope, receive, send)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/routing.py", line 288, in handle
    |     await self.app(scope, receive, send)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/routing.py", line 76, in app
    |     await wrap_app_handling_exceptions(app, request)(scope, receive, send)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    |     raise exc
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
    |     await app(scope, receive, sender)
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/routing.py", line 73, in app
    |     response = await f(request)
    |                ^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/fastapi/routing.py", line 301, in app
    |     raw_response = await run_endpoint_function(
    |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/fastapi/routing.py", line 214, in run_endpoint_function
    |     return await run_in_threadpool(dependant.call, **values)
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/starlette/concurrency.py", line 39, in run_in_threadpool
    |     return await anyio.to_thread.run_sync(func, *args)
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/anyio/to_thread.py", line 56, in run_sync
    |     return await get_async_backend().run_sync_in_worker_thread(
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 2505, in run_sync_in_worker_thread
    |     return await future
    |            ^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/.venv/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 1005, in run
    |     result = context.run(func, *args)
    |              ^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/keep/api/routes/preset.py", line 475, in get_preset_alerts
    |     preset_alerts = search_engine.search_alerts(preset_dto.query)
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/keep/searchengine/searchengine.py", line 161, in search_alerts
    |     filtered_alerts = self._search_alerts_by_sql(
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/keep/searchengine/searchengine.py", line 138, in _search_alerts_by_sql
    |     filtered_alerts = self.elastic_client.search_alerts(
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/home/runner/work/keep/keep/keep/api/core/elastic.py", line 1[98](https://github.com/keephq/keep/actions/runs/15237037962/job/42853209989?pr=4904#step:8:99), in search_alerts
    |     raise Exception(f"Failed to run query in Elastic: {e}")
    | Exception: Failed to run query in Elastic: BadRequestError(400, 'verification_exception', 'Found 1 problem\nline 1:44: Unknown column [lastReceived]')
    +------------------------------------
```

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -18,6 +18,7 @@ from sqlalchemy.orm import sessionmaker
 from sqlalchemy.pool import StaticPool
 from sqlmodel import Session, SQLModel, create_engine
 from starlette_context import context, request_cycle_context
+from playwright.sync_api import Page
 
 # This import is required to create the tables
 from keep.api.core.dependencies import SINGLE_TENANT_UUID
@@ -559,6 +560,19 @@ def browser():
         yield page
         context.close()
         browser.close()
+
+
+@pytest.fixture
+def auth_page(browser: Page):
+    """Fresh page with clean cookies for auth tests"""
+    # Get the browser from the page
+    browser_instance = browser.context.browser
+    if not browser_instance:
+        raise ValueError("Browser instance not found")
+    auth_context = browser_instance.new_context()
+    auth_page = auth_context.new_page()
+    yield auth_page
+    auth_context.close()
 
 
 def _create_valid_event(d, lastReceived=None):

--- a/tests/e2e_tests/test_end_to_end_db_auth.py
+++ b/tests/e2e_tests/test_end_to_end_db_auth.py
@@ -3,18 +3,18 @@ from playwright.sync_api import Page
 from tests.e2e_tests.utils import save_failure_artifacts
 
 
-def test_start_with_keep_db(browser: Page, setup_page_logging, failure_artifacts):
+def test_start_with_keep_db(auth_page: Page, setup_page_logging, failure_artifacts):
     # Navigate to signin page
-    browser.goto("http://localhost:3001/signin")
+    auth_page.goto("http://localhost:3001/signin")
     # Wait for the page to load
-    browser.wait_for_selector("text=Sign in")
+    auth_page.wait_for_selector("text=Sign in")
     # Fill in credentials
-    browser.get_by_placeholder("Enter your username").fill("keep")
-    browser.get_by_placeholder("Enter your password").fill("keep")
+    auth_page.get_by_placeholder("Enter your username").fill("keep")
+    auth_page.get_by_placeholder("Enter your password").fill("keep")
     # Click sign in and wait for navigation
-    browser.get_by_role("button", name="Sign in").click()
+    auth_page.get_by_role("button", name="Sign in").click()
     try:
-        browser.wait_for_url("http://localhost:3001/incidents", timeout=10000)
+        auth_page.wait_for_url("http://localhost:3001/incidents", timeout=10000)
     except Exception:
-        save_failure_artifacts(browser)
+        save_failure_artifacts(auth_page)
         raise

--- a/tests/e2e_tests/test_pushing_prometheus_alerts.py
+++ b/tests/e2e_tests/test_pushing_prometheus_alerts.py
@@ -106,8 +106,9 @@ def test_pulling_prometheus_alerts_to_provider(
 
             # Try to get to the Feed page
             feed_link = browser.get_by_role("link", name="Feed")
-            feed_link.wait_for(state="visible")
-            feed_link.click(timeout=10000)  # Increase timeout to 10 seconds
+            feed_link.click()
+
+            browser.wait_for_url("**/alerts/feed")
 
             # Wait for alerts to load with increased timeout
             alert_element = browser.wait_for_selector(
@@ -151,8 +152,8 @@ def test_pulling_prometheus_alerts_to_provider(
         )
 
         providers_link = browser.get_by_role("link", name="Providers")
-        providers_link.wait_for(state="visible")
-        providers_link.click(timeout=10000, force=True)  # Use force if needed
+        providers_link.click()
+        browser.wait_for_url("**/providers")
 
     except Exception as e:
         print(f"Failed to click Providers link: {e}")

--- a/tests/test_enrichments.py
+++ b/tests/test_enrichments.py
@@ -458,6 +458,9 @@ def test_enrichment_with_elastic(
         json=mock_alert_dto.dict(),
     )
 
+    # wait for the alert to be indexed
+    sleep(1)
+
     # now query the feed preset to get the alerts
     response = client.get(
         "/preset/feed/alerts",

